File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
       int left = 0;
       int right = heights.size ()-1;
        int res =0;
       while(left<right){
            int curr = (right-left)*min(heights[left],heights[right]);
            res = max(res,curr);

            if(heights[left] <= heights[right]){
                left++;
            }
            else{
                right--;
            }
       } 
       return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        mySet = set()
        l , r = 0 , 0
        res = 0
        while r<len(s):
            while s[r] in mySet:
                mySet.remove(s[l])
                l+=1
            res = max(res, r-l+1)
            mySet.add(s[r])
            r+=1
        return res

